components/patient-data/impl/src/main/java/org/phenotips/data/internal/PatientXWikiConsentManager.java

```diff
@@ -0,0 +1,363 @@
+    {
+        JSONArray json = new JSONArray();
+        for (Consent consent : consents) {
+            json.add(consent.toJson());
+        }
+        return json;
+    }
+
+    @Override public List<Consent> fromJson(JSON json)
+    {
+        return null;
+    }
+
+    private List<Consent> loadConsentsFromSystem()
+    {
+        List<Consent> consents = new LinkedList<>();
+        try {
+            DocumentReference configDocRef = referenceResolver.resolve(this.configurationPageReference);
+            DocumentModelBridge configDocBridge = bridge.getDocument(configDocRef);
+            XWikiDocument configDoc = (XWikiDocument) configDocBridge;
+            List<BaseObject> consentObjects = configDoc.getXObjects(consentReference);
+            if (consentObjects != null) {
+                for (BaseObject consentObject : consentObjects) {
+                    consents.add(this.fromXWikiConsentConfiguration(consentObject, configDoc));
+                }
+            }
+        } catch (Exception ex) {
+            /* if configuration cannot be loaded, it cannot be loaded; nothing to be done */
+            logger.error("Could not load the configurations for patient consents. {}", ex.getMessage());
+        }
+        return consents;
+    }
+
+    // fixme. must be run on every save of XWikiPreferences. There is no UI yet, however if there is to be one, that
+    // must be a implemented.
+    private void refreshSystemConsents()
+    {
+        this.systemConsents = loadConsentsFromSystem();
+    }
+
+    private Consent fromXWikiConsentConfiguration(BaseObject xwikiConsent, XWikiDocument configDoc)
+    {
+        String id = xwikiConsent.getStringValue("id");
+        String description = configDoc.display("description", "view", xwikiConsent, contextProvider.get());
+        /* removing divs */
+        description = cleanDescription(description);
+        boolean required = intToBool(xwikiConsent.getIntValue("required"));
+        return new DefaultConsent(id, description, required);
+    }
+
+    private static String cleanDescription(String toClean)
+    {
+        String clean = toClean;
+        clean = clean.replace("<div>", "").replace("</div>", "");
+        clean = clean.replace("<p>", "").replace("</p>", "");
+        clean = clean.replaceAll("[{]{2}(/{0,1})html(.*?)[}]{2}", "");
+        return clean;
+    }
+
+    /**
+     * @param grant if true will grant the consent, otherwise will revoke
+     * @return if operation was successful
+     */
+    private boolean manageConsent(Patient patient, String consentId, boolean grant)
+    {
+        if (!this.isValidId(consentId)) {
+            this.logger.warn("Invalid consent id ({}) was supplied", consentId);
+            return false;
+        }
+        try {
+            SaveablePatientConsentHolder consentHolder = this.getPatientConsentHolder(patient);
+            List<String> currentConsents = consentHolder.getConsents();
+            if (grant) {
```

Comment:
nope, it can't be. The ```else``` clause should be executed iff ```grant``` is ```false```, with no other conditions.